### PR TITLE
feat: vectorize Nx.compatible?

### PR DESCRIPTION
**Pull request by polvalente**

### CODE REVIEW

File: nx/lib/nx.ex [hunk 1]
```diff
@@ -3586,14 +3586,53 @@ defmodule Nx do
       iex> Nx.compatible?(%{foo: Nx.iota({3, 2})}, %{bar: Nx.iota({3, 2})})
       false
 
+  ## Vectorized tensors
+
+  Same compatibility criteria applies to vectorized tensors, but there's
+  the additional requirement that if a vectorized axis name appears in both
+  tensors, it must appear with the same size, or one of them must be 1.
```
Comment (josevalim) [inline nx/lib/nx.ex:3593]: Vectorized axes must be the same size. We don't consider shape `{1}` to be compatible with `{10}`.

File: nx/lib/nx.ex [hunk 2]
```diff
@@ -3586,14 +3586,53 @@ defmodule Nx do
...
   def compatible?(left, right)
 
-  def compatible?(%T{} = left, %T{} = right) do
-    Nx.Shared.raise_vectorized_not_implemented_yet(left, __ENV__.function)
-    Nx.Shared.raise_vectorized_not_implemented_yet(right, __ENV__.function)
+  def compatible?(
+        %T{type: type, shape: shape, names: l_names, vectorized_axes: l_axes},
+        %T{type: type, shape: shape, names: r_names, vectorized_axes: r_axes}
+      )
+      when l_axes != [] or r_axes != [] do
```
Comment (josevalim) [inline nx/lib/nx.ex:3610]: I think this should pretty much be `l_axes == r_axes`.

File: nx/lib/nx.ex [hunk 3]
```diff
@@ -3586,14 +3586,31 @@ defmodule Nx do
...
   def compatible?(left, right)
 
-  def compatible?(%T{} = left, %T{} = right) do
-    Nx.Shared.raise_vectorized_not_implemented_yet(left, __ENV__.function)
-    Nx.Shared.raise_vectorized_not_implemented_yet(right, __ENV__.function)
+  def compatible?(
+        %T{type: type, shape: shape, names: l_names, vectorized_axes: l_axes},
+        %T{type: type, shape: shape, names: r_names, vectorized_axes: r_axes}
+      ) do
+    l_axes == r_axes and compatible_names?(l_names, r_names)
```
Comment (josevalim) [inline nx/lib/nx.ex:3610]:
```suggestion
        %T{type: type, shape: shape, names: l_names, vectorized_axes: v_axes},
        %T{type: type, shape: shape, names: r_names, vectorized_axes: v_axes}
      ) do
    compatible_names?(l_names, r_names)
```

**Review by josevalim — APPROVED** [review]

File: nx/lib/nx.ex (hunk 3 above)
Comment (polvalente) [reply nx/lib/nx.ex:3610]: This doesn't work because if axes are different, there's no clause that catches this

**Pull request merged**